portal-service/src/com/liferay/portal/model/PortletInstance.java

```diff
@@ -114,6 +133,48 @@ public String toString() {
 		return getPortletInstanceKey();
 	}
 
+	private static FullInstanceId _buildFullInstanceId(String fullInstanceId) {
+		if (fullInstanceId == null) {
+			throw new InvalidParameterException(
+				"The fullInstanceId must not be null.");
+		}
+
+		String[] tokens = StringUtil.split(
+			fullInstanceId, StringPool.UNDERLINE);
+
+		int count = StringUtil.count(fullInstanceId, StringPool.UNDERLINE);
```

Comment:
I'm not sure I understand the answer. Let's see what are the values in each of those cases:
abc_abc
tokens.length=2
count=1

abc_abc_
tokens.length=3 (not fully sure about this one)
count=2

I don't see why the count is needed. And it's error prone to count the same thing twice in two different ways.